src/main/scala/org/globalforestwatch/summarystats/gfwpro_dashboard/GfwProDashboardAnalysis.scala

```diff
@@ -45,13 +43,20 @@ object GfwProDashboardAnalysis extends SummaryAnalysis {
             spark: SparkSession,
             kwargs: Map[String, Any]): Unit = {
 
-    val enrichedRDD = intersectWithContextualLayer(featureRDD, kwargs, spark)
+    // only create this once for all analyses, analyze() can be slow
+    val spatialFeatureRDD = new SpatialRDD[GeoSparkGeometry]()
+    spatialFeatureRDD.rawSpatialRDD = featureRDD
+      .map(fromGeotrellisFeature[Geometry, FeatureId, GeoSparkGeometry])
+      .toJavaRDD()
+    spatialFeatureRDD.analyze()
+
+    val enrichedRDD = intersectWithContextualLayer(spatialFeatureRDD, kwargs, spark)
 
     val summaryRDD: RDD[(FeatureId, GfwProDashboardSummary)] =
       GfwProDashboardRDD(enrichedRDD, GfwProDashboardGrid.blockTileGrid, kwargs)
 
     val fireCount: RDD[(FeatureId, GfwProDashboardDataDateCount)] =
-      GfwProDashboardAnalysis.fireStats(featureType, spark, kwargs)
+      GfwProDashboardAnalysis.fireStats(spatialFeatureRDD, featureType, spark, kwargs)
```

Comment:
We actually need to use the enrichedRDD here. We later join the summaryRDD and FireCountRDD using the FeatureID.

In the enriched RDD, we change the FeatureID from GwfProFeatureId to CombineFeatureId[GfwProFeatureId, GADMFeatureId]

Reply:
@thomas-maschler oh I see, so I should return an RDD of CombineFeatureId[GfwProFeatureId, GADMFeatureId] from this analysis?